data_diff/diff_tables.py

```diff
@@ -227,6 +244,24 @@ def diff_tables(self, table1: TableSegment, table2: TableSegment) -> DiffResult:
         if self.bisection_factor < 2:
             raise ValueError("Must have at least two segments per iteration (i.e. bisection_factor >= 2)")
 
+        table1, table2 = self._threaded_call("with_schema", [table1, table2])
+        for c in table1._relevant_columns:
```

Comment:
Yes!!!!! ❤️ 🚀 

This is such an incredible seed for a lot of beautiful warnings and exceptions as we find 'incompatibilities' between the schemas. We'll (you especially) will become _experts_ in all of the incompatibilities between various types, between various databases. For some people who are doing a migration, the wealth of knowledge encoded in the warnings might be _more_ valuable to some than the actual checking! This will be gold. 🏅 

Might be worth throwing into its own method as you polish this up for merge.